Runtime panic error for the operator when a resource is created without storage

<!-- Please answer these questions before submitting your bug report. Thanks! -->

**What version of redis operator are you using?**

redis-operator version: 0.16.0

**Does this issue reproduce with the latest release?**
Yes


**What operating system and processor architecture are you using (`kubectl version`)?**

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version
Client Version: v1.29.3
</pre></details>

**What did you do?**
I created a Redis standalone resource without defining the storage resource.

<!--
If possible, provide a recipe for reproducing the error.
A detailed sequence of steps describing what to do to observe the issue is good.
A complete runnable bash shell script is best.
-->



**What did you expect to see?**
Should provision the Redis instance with default config


**What did you see instead?**
Redis operator is in a crash loopback.
```
{"level":"info","ts":"2024-05-10T08:46:58Z","msg":"Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference","controller":"redis","controllerGroup":"redis.redis.opstreelabs.in","controllerKind":"Redis","Redis":{"name":"redis-standalone","namespace":"integration"},"namespace":"integration","name":"redis-standalone","reconcileID":"7fa7d9ba-823f-4d01-9b79-5de377f5dae9"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x17854ba]
```
**Additional context**
The issue was fixed for the Redis cluster 3 days ago via https://github.com/OT-CONTAINER-KIT/redis-operator/pull/887, we probably need to implement a similar solution for redis standalone and redis replication.

--- a/k8sutils/finalizer.go
+++ b/k8sutils/finalizer.go
@@ -25,7 +25,7 @@ const (
 func HandleRedisFinalizer(ctrlclient client.Client, k8sClient kubernetes.Interface, logger logr.Logger, cr *redisv1beta2.Redis) error {
 	if cr.GetDeletionTimestamp() != nil {
 		if controllerutil.ContainsFinalizer(cr, RedisFinalizer) {
-			if !cr.Spec.Storage.KeepAfterDelete {
+			if cr.Spec.Storage != nil && !cr.Spec.Storage.KeepAfterDelete {
 				if err := finalizeRedisPVC(k8sClient, logger, cr); err != nil {
 					return err
 				}
@@ -44,7 +44,7 @@ func HandleRedisFinalizer(ctrlclient client.Client, k8sClient kubernetes.Interfa
 func HandleRedisClusterFinalizer(ctrlclient client.Client, k8sClient kubernetes.Interface, logger logr.Logger, cr *redisv1beta2.RedisCluster) error {
 	if cr.GetDeletionTimestamp() != nil {
 		if controllerutil.ContainsFinalizer(cr, RedisClusterFinalizer) {
-			if !cr.Spec.Storage.KeepAfterDelete {
+			if cr.Spec.Storage != nil && !cr.Spec.Storage.KeepAfterDelete {
 				if err := finalizeRedisClusterPVC(k8sClient, logger, cr); err != nil {
 					return err
 				}
@@ -63,7 +63,7 @@ func HandleRedisClusterFinalizer(ctrlclient client.Client, k8sClient kubernetes.
 func HandleRedisReplicationFinalizer(ctrlclient client.Client, k8sClient kubernetes.Interface, logger logr.Logger, cr *redisv1beta2.RedisReplication) error {
 	if cr.GetDeletionTimestamp() != nil {
 		if controllerutil.ContainsFinalizer(cr, RedisReplicationFinalizer) {
-			if !cr.Spec.Storage.KeepAfterDelete {
+			if cr.Spec.Storage != nil && !cr.Spec.Storage.KeepAfterDelete {
 				if err := finalizeRedisReplicationPVC(k8sClient, logger, cr); err != nil {
 					return err
 				}

